content/post/lessons-learned-building-an-accessible-web-application-framework/index.md

```diff
@@ -0,0 +1,68 @@
+
+## What does it mean for a site to be accessible
+
+I'll be honest, when we started in 2017, I thought creating an accessible website simply meant appropriately applying ARIA labels to DOM elements. In the last few years, we've learned, with the help of our accessibility researchers, that creating an accessible site is so much more than that. So, let's hop in our time machine and discuss how we should have approached accessibility with appropriate hindsight (foresight? Time travel is confusing).
+
+### Identify how users interact with our site with accessible technologies
+
+As we discussed above, our broad user base includes a diverse array of people and an equally diverse array of accessibility needs. When discussing web accessibility, it's easy to assume that means ensuring a screen reader appropriately can navigate and read your site. Screen reader compatibility is important but if your user has impaired mobility, but fine eyesight, you're not really helping them. We've compiled the following use cases to ensure we are covering all expected interactions and needs.
+
+* Impaired vision
+* Impaired mobility
+* Impaired cognition
+
+Additionally, individuals could temporarily fall into any of the above categories through injury, device capability, location, or conditions.
+
+### Create areas of focus for each use case
+
+For each of the above use cases we've identified the appropriate assistive technologies to focus on.
+
+#### Impaired vision
+
+* Focus on helping the user access the site content by ensuring appropriate contrast ratios, font sizing, spacing, color usage, and screen reader support. A good way to get a feel for this user's experience is to enable a screen reader (all Macs have voice-over), close your eyes, and navigate your site. Does it make sense?
+
+#### Impaired mobility
+
+* Focus on helping the user navigate the site using devices other than a cursor by enabling keyboard navigation and supplying redundant information access. For example, providing alternative methods to take actions that require gestures on mobile.
+
+#### Impaired cognition
+
+* Focus on applying layout simplicity to our workflows and components as well as improving keyword comprehension by emphasizing common design patterns and expected behaviors. For example, buttons should not navigate a user to a new page, that is the role of Links.
+
+#### All impairments
+
+* Provide a consistent semantic layout and access to information regardless of form factor (desktop, mobile).
```

Comment:
Can you provide an example of this with desktop and mobile?

Reply:
added